testing/tests/DevExpress.ui.widgets/treeViewParts/checkboxes.js

```diff
@@ -213,6 +213,87 @@ QUnit.test('T195986', function(assert) {
     assert.strictEqual($(checkboxes[0]).dxCheckBox('instance').option('value'), true);
 });
 
+['none', 'normal', 'selectAll'].forEach((showCheckBoxesMode) => {
+    ['multiple', 'single'].forEach((selectionMode) => {
+        [false, true].forEach((selectNodesRecursive) => {
```

Comment:
Also, there is the https://js.devexpress.com/Documentation/21_1/ApiReference/UI_Components/dxTreeView/Configuration/#selectByClick option but I am not sure that we have to add tests for the same selection events in this PR